### PR TITLE
Refactor:  separate delete and insert account index stats

### PR DESCRIPTION
**Pull request by HaoranYi**
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/bucket_map_holder.rs [hunk 1]
```diff
@@ -446,7 +446,7 @@ pub mod tests {
         let bins = 1;
         let test = BucketMapHolder::<u64>::new(bins, &Some(AccountsIndexConfig::default()), 1);
         let threads = 2;
-        let time = AGE_MS * 5 / 2;
+        let time = AGE_MS * 8 / 3;
```
Comment (brooksprumo) [inline runtime/src/bucket_map_holder.rs:449]: Why did this change?

**Review by brooksprumo — COMMENTED** [review]
Thanks, I like this direction; I find these new functions more intuitive to reason about. Left one comment for my understanding.

File: runtime/src/bucket_map_holder.rs (hunk 1 above)
Comment (HaoranYi) [reply runtime/src/bucket_map_holder.rs:449]: https://buildkite.com/solana-labs/solana/builds/75341#01813b58-9c10-4e40-84a4-fccda52386bb
To help fixing the intermittent test failure, like the one above.

Comment (brooksprumo) [reply runtime/src/bucket_map_holder.rs:449]: Gotcha. Can you pull this out into a separate PR since it sounds unrelated to the title of this PR?

Comment (HaoranYi) [reply runtime/src/bucket_map_holder.rs]: Sure.

**Pull request merged**